### PR TITLE
Decrease vertical scroll speed by using deltaMode

### LINKED ISSUE #149: Browser inconsistencies in scrollY because deltaMode not used calculate wheelEvent deltas
Part of the solution to https://github.com/emscripten-core/emscripten/issues/6283 appears to be to use `wheelEvent->deltaMode` in [Emscripten_HandleWheel](https://github.com/emscripten-ports/SDL2/blob/6a72595649456d1aaa8d15e152c830aab1c4c422/src/video/emscripten/SDL_emscriptenevents.c#L412) to calculate the deltaY. 

The conversions used [in a fix](https://github.com/emscripten-core/emscripten/pull/7968) for libraries using [library_browser.js](https://github.com/emscripten-core/emscripten/blob/main/src/library_browser.js) instead of [library_html5.js](https://github.com/emscripten-core/emscripten/blob/main/src/library_html5.js) are:

```
case 0:
  // DOM_DELTA_PIXEL: 100 pixels make up a step
  delta /= 100;
  break;
case 1:
  // DOM_DELTA_LINE: 3 lines make up a step
  delta /= 3;
  break;
case 2:
  // DOM_DELTA_PAGE: A page makes up 80 steps
  delta *= 80;
  break;
```

Does adding this logic here make sense, should I prep a PR?

### PR DESCRIPTION
**Pull request by thomasballinger**
Resolves https://github.com/emscripten-ports/SDL2/issues/149

This works for me (it makes scrolling interfaces that were very hard to use before usable!) but I don't know how to test. This would be a major behavior change for anything using scrolling in SDL2, but should be more consistent with native SDL2.

### CODE REVIEW

File: src/video/emscripten/SDL_emscriptenevents.c [hunk 1]
```diff
@@ -408,8 +408,24 @@ Emscripten_HandleMouseFocus(int eventType, const EmscriptenMouseEvent *mouseEven
...
+          // DOM_DELTA_LINE: 3 lines make up a step
+          deltaY /= 3;
+          break;
+        case 2:
+          // DOM_DELTA_PAGE: A page makes up 80 steps
+          deltaY *= 80;
+          break;
+    }
+
     SDL_WindowData *window_data = userData;
```
Comment (Daft-Freak) [inline src/video/emscripten/SDL_emscriptenevents.c:427]: Variable declarations should be at the top.

**Review by Daft-Freak — COMMENTED** [review]
Also, commit should be prefixed with `emscripten: ` (Both comments are so I can just push this upstream).

I guess all this needs for testing is checking that it behaves across borwsers/platforms. (Not that native SDL is even consistent across platforms...)

**Comment by thomasballinger** [comment]
One way to test is to make these changes to `emsdk/upstream/emscripten/tools/ports/sdl2.py`:

`rm -rf emsdk/upstream/emscripten/cache/ports/sdl2.zip emsdk/upstream/emscripten/cache/ports/sdl2`

(updated)
```diff
8,9c8,9
< TAG = 'version_24'
< HASH = '5a8181acdcce29cdda7e7a4cc876602740f5b9deebd366ecec71ae15c4bbf1f352da4dd0e3c5e0ba8160709dda0270566d64a6cd3892da894463ecf8502836aa'
---
> TAG = 'scroll-speed'
> HASH = '9d3c51eb4c77bd89684b21e8f342bcc290646e01aafa9ba62d0473585d834a02ca397f652c723140b4f9f53976552eed89feeec190ea1422fde770da11512549'
12a13
> 
23c24
<   ports.fetch_project('sdl2', 'https://github.com/emscripten-ports/SDL2/archive/' + TAG + '.zip', SUBDIR, sha512hash=HASH)
---
>   ports.fetch_project('sdl2', 'https://github.com/thomasballinger/SDL2/archive/refs/heads/scroll-speed.zip', SUBDIR, sha512hash=HASH)

```

I've tried this on Firefox and Chrome, and both feel much better to me.

File: src/video/emscripten/SDL_emscriptenevents.c [hunk 2]
```diff
@@ -409,7 +409,25 @@ static EM_BOOL
 Emscripten_HandleWheel(int eventType, const EmscriptenWheelEvent *wheelEvent, void *userData)
 {
     SDL_WindowData *window_data = userData;
-    SDL_SendMouseWheel(window_data->window, 0, (float)wheelEvent->deltaX, (float)-wheelEvent->deltaY, SDL_MOUSEWHEEL_NORMAL);
+
+    float deltaY = wheelEvent->deltaY;
+
+    switch (wheelEvent->deltaMode) {
+        case 0:
```
Comment (sbc100) [inline src/video/emscripten/SDL_emscriptenevents.c:416]: Why not actually use `DOM_DELTA_PIXEL` et al from `emscripten/html5.h` here in the switch?

**Review by sbc100 — APPROVED** [review]
Looks good to me but maybe wait for @Daft-Freak to take a look.

File: src/video/emscripten/SDL_emscriptenevents.c [hunk 3]
```diff
@@ -409,7 +409,22 @@ static EM_BOOL
 Emscripten_HandleWheel(int eventType, const EmscriptenWheelEvent *wheelEvent, void *userData)
 {
     SDL_WindowData *window_data = userData;
-    SDL_SendMouseWheel(window_data->window, 0, (float)wheelEvent->deltaX, (float)-wheelEvent->deltaY, SDL_MOUSEWHEEL_NORMAL);
+
+    float deltaY = wheelEvent->deltaY;
+
+    switch (wheelEvent->deltaMode) {
+        case DOM_DELTA_PIXEL:
+          deltaY /= 100; // 100 pixels make up a step
```
Comment (Daft-Freak) [inline src/video/emscripten/SDL_emscriptenevents.c:417]: Sorry, missed this earlier but I believe upstream prefers `/*c-style comments*/`. Also indentation looks a little off.

**Review by Daft-Freak — CHANGES_REQUESTED** [review]
I haven't tested this, but it looks sane. (Any handy test code around?)

File: src/video/emscripten/SDL_emscriptenevents.c (hunk 3 above)
Comment (thomasballinger) [reply src/video/emscripten/SDL_emscriptenevents.c:417]: whoops, thanks

**Comment by thomasballinger** [comment]
Here's some very minimal test code that console.logs scroll.y. It has only been tested with Emscripten. https://gist.github.com/thomasballinger/825b74b1fb39ca37baf2b537e266ac68

**Comment by Daft-Freak** [comment]
Okay, every browser I tried used pixels, and they all had different delta values.

```
Firefox (Linux): +/-114
Chrome (Linux): +/-106
Firefox (Windows): +/-106
Chrome (Windows): +/-200
```

After applying this I get this for moving the wheel down three times and then up three times: 
```
wheel.y:  -1
wheel.y:  -1
wheel.y:  -1
wheel.y:  0
wheel.y:  1
wheel.y:  2
```
Except in Windows Chrome, where it's consistently +/-2.

Native Linux/Windows seems to be:
```
wheel.y -1
wheel.y -1
wheel.y -1
wheel.y 1
wheel.y 1
wheel.y 1
```
So that looks about right. It does result in mostly 0 deltas when using my touchpad, but native does that too.

**Review by Daft-Freak — APPROVED** [review]

**Comment by thomasballinger** [comment]
Let me know if there's anything left I can help with here!

**Pull request merged**